Expensify/App
# Clean up policy actions

cc @roryabraham Longer term we probably want to solve the issue of why we can't get policies and create them at the same time, but this feels like a logical refactor. I think by splitting these out into their own methods things feel more intentional - even if the behavior is exactly the same for now.

### Details
Clean up Policy actions

### Fixed Issues
$ https://github.com/Expensify/App/issues/6153

### Tests / QA Steps
1. Log out of NewDot
2. Sign up for a new account via OldDot on mobile web with a private domain
2. Choose "Set up my company for free"
3. Verify that the page opens to NewDot and the workspace is visible
4. Close the workspace and validate the email
5. Sign into OldDot with the same account (desktop web) and navigate to policies. Select the workspace.
6. Verify the page opens to NewDot with the workspace visible.
7. Sign out of NewDot and repeat steps 5 & 6

### Tested On


### Screenshots


#### Web


#### Mobile Web


#### Desktop


#### iOS


#### Android
